mat64/vector.go

```diff
@@ -357,6 +360,61 @@ func (v *Vector) MulVec(a Matrix, b *Vector) {
 	}
 }
 
+// MarshalBinary encodes the receiver into a binary form and returns the result.
```

Comment:
Do you think it's worth moving this code and the similar code for *Dense into io.go? This way all of the constants are in closer view.